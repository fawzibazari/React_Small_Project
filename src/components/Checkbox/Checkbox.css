/* .Checkbox {
  height: 7vh;
    width: 2vw;
    
} */
:root {
  --color-checked: rgb(33, 137, 247);
  --color-border: rgb(61, 63, 65);
}
.checkbox .border {
  position: relative;
border: 2px solid var(--color-border);
border-radius: 4px;
margin-right: 8px;
width: 20px;
height: 20px;   
margin-top: 14px; 
}

.checkbox .indicator {
  position: absolute;
z-index: 1;
background: var(--color-checked);
width: 20px;
height: 20px;
transform: scale(0);
transition: transform 100ms;
}
.checkbox .indicator.checked {
transform: scale(1);
    
}