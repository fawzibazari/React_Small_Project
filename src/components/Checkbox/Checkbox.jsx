import PropTypes from 'prop-types';
import React from 'react';
import "./Checkbox.css";

// import style from "./Checkbox.module.css";

export default function CheckBox({ checked, onChange}) {
  
  return (
    <div className="checkbox">
    <div className="border" onClick={() => onChange(!checked)}>
    <div data-testid="checkbox" className={`indicator ${checked ? "checked" : ""}`} />
      </div>
    </div>
  )
}
CheckBox.propTypes = {
  checked: PropTypes.bool,
  onChange: PropTypes.func
};