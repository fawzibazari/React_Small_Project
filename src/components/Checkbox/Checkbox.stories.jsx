import Checkbox from './Checkbox'
export default {
    title: 'component/Checkbox',
    component : 'Checkbox'
}

export const Checked = () => <Checkbox checked={true} onChange={() => {}}/>
export const UnChecked = () => <Checkbox checked={false} onChange={() => {}}/>
export const CheckBox = () => <Checkbox onChange={() => {}}/>