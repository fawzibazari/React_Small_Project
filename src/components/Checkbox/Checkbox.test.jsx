import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Checkbox from "./Checkbox";
import "./Checkbox.css";

test("print hello world to the user", () => {
    render(<Checkbox/>);

    const element = screen.getByTestId("checkbox");
    expect(element).toMatchSnapshot();
})
test("changement du classname au click", () => {
    render(<Checkbox/>);

    const element = screen.getByTestId("checkbox");
    expect(element).toBeInTheDocument();
    userEvent.click(element);
    expect(element).toHaveClass("indicator");
})