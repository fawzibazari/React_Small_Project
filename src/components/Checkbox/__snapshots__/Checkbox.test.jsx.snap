// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print hello world to the user 1`] = `
<div
  class="checkbox"
  data-testid="checkbox"
>
  <div
    class="border"
  >
    <div
      class="indicator "
    />
  </div>
</div>
`;
